File: _study/COS_PRO_2/cpp/stage5/5.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(int a, int b) {
    int answer = 0;

    for(int i = 1; i <= b; i++) {
        if((b * i) % a == 0) {
            answer = b * i;
            break;
        }
    }

    return answer;
}

// 아래는 테스트케이스 출력을 해보기 위한 main 함수입니다. 아래에는 잘못된 부분이 없으니, 위의 코드만 수정하세요.
int main() {
    int a = 4;
    int b = 6;
    int ret = solution(a, b);

    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    cout << "solution 함수의 반환 값은 " << ret << "입니다." << endl;
}

File: _study/COS_PRO_1/python/stage1/5.py
def solution(n):
    answer = 0
    lst = list(range(n*n, 0, -1)) # 뒤집어진 리스트생성
    if n % 2 == 0: # n이 짝수이면
        check = True
        cnt = 1
        inc = 1
        for i in range(n):
            answer += lst[cnt]
            if check == False:
                inc += 4
                cnt += inc
                check = True
            else:
                if inc == 1:
                    inc += 1
                cnt += inc
                check = False
    else:
        check = False
        cnt = 0
        inc = 4
        for i in range(n):
            answer += lst[cnt]
            if check == False:
                cnt += inc
                check = True
            else:
                cnt += inc
                inc += 4
                check = False
    return answer


n1 = 5
ret1 = solution(n1)

print("solution 함수의 반환 값은", ret1, "입니다.")

n2 = 6
ret2 = solution(n2)

print("solution 함수의 반환 값은", ret2, "입니다.")
